pyproject.toml

```diff
@@ -1,13 +1,82 @@
 [build-system]
-requires = ["setuptools", "wheel", "setuptools-rust"]
+requires = ["maturin>=0.15,<0.16"]
```

Comment:
Curious what the issues are with newer versions of `maturin`. What happens when trying 0.16 (or newer)?

Also wondering if we can move to [1.1.0]( https://github.com/PyO3/maturin/releases/tag/v1.1.0 ) (where maybe things are more stable?)

Reply:
Did this mostly to align with [ADP's maturin constraint](https://github.com/apache/arrow-datafusion-python/blob/1fde8e47ae6714f02c2d2a088577e92e814dfa34/pyproject.toml#L19), though I could try bumping us to a more recent version to see how that impacts things.

Will follow up on https://github.com/dask-contrib/dask-sql/pull/1181#discussion_r1278100781 accordingly depending on how that goes